hpi-archetype/pom.xml

```diff
@@ -69,14 +69,22 @@
       <url>http://repo.jenkins-ci.org/public/</url>
     </pluginRepository>
   </pluginRepositories>
-  <!-- If you want to depend on other plugins:
   <dependencies>
+    <!-- Remove this dependency if you don't need a Pipeline step -->
+    <dependency>
+      <groupId>org.jenkins-ci.plugins.workflow</groupId>
+      <artifactId>workflow-step-api</artifactId>
+      <version>2.0</version>
+      <optional>true</optional>
```

Comment:
:-1: on optional dependencies. This one adds no GUI or large library dependencies—it is intended to be cheap.